analytics/src/main/java/com/segment/analytics/internal/Utils.java

```diff
@@ -298,25 +298,12 @@ public static String assertNotNullOrEmpty(String text, @Nullable String name) {
         return Collections.unmodifiableList(new ArrayList<>(list));
     }
 
-    /** Creates a unique device id. */
-    public static String getDeviceId() {
-        // unique id generated from DRM API
-        String uniqueID = getUniqueID();
-        if (!isNullOrEmpty(uniqueID)) {
-            return uniqueID;
-        }
-
-        // If this still fails, generate random identifier that does not persist across
-        // installations
-        return UUID.randomUUID().toString();
-    }
-
     /**
      * Workaround for not able to get device id on Android 10 or above using DRM API {@see
      * https://stackoverflow.com/questions/58103580/android-10-imei-no-longer-available-on-api-29-looking-for-alternatives}
      * {@see https://developer.android.com/training/articles/user-data-ids}
      */
-    private static String getUniqueID() {
+    public static String getUniqueID() {
```

Comment:
given that this is not a safe API to use anymore, would it make sense to move it into the `GetDeviceIdTask` class?